### PR TITLE
Add support for `Task`/`Workflow` hooks and create a `GlobalConfig`

### PR DESCRIPTION
**Pull request by flaviuvadan**
Fixes: #258 

CC: @JacobHayes (OP author)
CC: @elliotgunton (supporter of global config, PR discussion)

### CODE REVIEW

File: tests/test_global_config.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        assert c.workflow_post_init_hooks == ()
+
+        def hook1(w: Workflow) -> None:
+            w.name = '123'
+
+        def hook2(w: Workflow) -> None:
+            w.service_account_name = "abc"
+
+        c.workflow_post_init_hooks = [hook1, hook2]
+        assert len(c.workflow_post_init_hooks) == 2
```
Comment (Trollgeir) [inline tests/test_global_config.py:83]: This test (and the same above) seems a bit redundant because it's impossible that the length isn't 2

**Review by Trollgeir — APPROVED** [review]
LGTM!

The only thing I don't really like is that we're going to squash all this. It should either have been seperate PRs, or we should support rebasing :)

File: tests/test_global_config.py (hunk 1 above)
Comment (flaviuvadan) [reply tests/test_global_config.py:86]: Expanded test for the condition in which someone adds hooks on top of existing ones, which is why the PR contained this check for 2 hooks

Comment (flaviuvadan) [reply tests/test_global_config.py:86]: Should have added the check in the first version of the PR.. that would have reduced confusion. Apologies :(

File: docs/examples/gpu.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ def do():
 
 
 with Workflow("gpu") as w:
-    gke_k80_gpu = {"cloud.google.com/gke-accelerator": "nvidia-tesla-k80"}
+    gke_k80_gpu = {"cloud.google.com/gke-accelerator": "nvidia-tesla-t4"}
```
Comment (flaviuvadan) [inline docs/examples/gpu.md:42]: K80s were silently deprecated by Google Cloud so switching examples to T4s

**Review by JacobHayes — COMMENTED** [review]
Super excited for these!

**Pull request merged**